sysl2/sysl/deps.go

```diff
@@ -0,0 +1,29 @@
+package main
+
+import (
+	"fmt"
+	"github.com/anz-bank/sysl/src/proto"
+)
+
+type AppElement struct {
+	Name     string
+	Endpoint string
+}
+
+type AppDependency struct {
+	Self      *AppElement
+	Target    *AppElement
+	Statement *sysl.Statement
+}
+
+func (dep *AppDependency) String() string {
+	return fmt.Sprintf("%s:%s:%s:%s", dep.Self.Name, dep.Self.Endpoint, dep.Target.Name, dep.Target.Endpoint)
+}
+
+func MakeAppDependency(self, target *AppElement, stmt *sysl.Statement) *AppDependency {
```

Comment:
Avoid trivial `Make…` functions.

Reply:
hmm. Is there a common standard when we can use Make... functions?